### ISSUE TITLE
FirestoreProperty on enums is not working

### ISSUE THREAD

**Issue by moander** [googleapis/google-cloud-dotnet#2720]
ArgumentException Unable to convert value of Gender

#### Environment details

  - Package name and version: 1.0.0-beta13

#### Steps to reproduce

```csharp
    [FirestoreData]
    public class Person
    {
        [FirestoreProperty]
        public string Name { get; set; }

        [FirestoreProperty]
        public Gender Gender { get; set; }
    }

    public enum Gender
    {
        Female,
        Male,
    }
```

**Comment by jskeet** [comment]
In the next week I intend to implement the custom serialisation code, so will probably do this as part of that. Thanks for reporting

**Comment by jskeet** [comment]
Available in version 1.0.0-beta14.
